scene/2d/light_2d.cpp

```diff
@@ -379,6 +388,24 @@ Rect2 PointLight2D::get_anchorable_rect() const {
 void PointLight2D::set_texture(const Ref<Texture2D> &p_texture) {
 	texture = p_texture;
 	if (texture.is_valid()) {
+#ifdef DEBUG_ENABLED
+		if (Object::cast_to<AnimatedTexture>(*p_texture)) {
+			WARN_PRINT(vformat("AnimatedTexture cannot be used as a PointLight2D texture (%s).", get_path()));
+		} else if (Object::cast_to<AtlasTexture>(*p_texture)) {
+			WARN_PRINT(vformat("AtlasTexture cannot be used as a PointLight2D texture (%s).", get_path()));
+		} else if (Object::cast_to<CameraTexture>(*p_texture)) {
+			WARN_PRINT(vformat("CameraTexture cannot be used as a PointLight2D texture (%s).", get_path()));
+		} else if (Object::cast_to<CanvasTexture>(*p_texture)) {
+			WARN_PRINT(vformat("CanvasTexture cannot be used as a PointLight2D texture (%s).", get_path()));
+		} else if (Object::cast_to<MeshTexture>(*p_texture)) {
+			WARN_PRINT(vformat("MeshTexture cannot be used as a PointLight2D texture (%s).", get_path()));
+		} else if (Object::cast_to<Texture2DRD>(*p_texture)) {
+			WARN_PRINT(vformat("Texture2DRD cannot be used as a PointLight2D texture (%s).", get_path()));
+		} else if (Object::cast_to<ViewportTexture>(*p_texture)) {
+			WARN_PRINT(vformat("ViewportTexture cannot be used as a PointLight2D texture (%s). As a workaround, assign the value returned by ViewportTexture's `get_image()` instead.", get_path()));
+		}
```

Comment:
I think this should be replaced by a method that takes texture type and automatically includes it in the warning message. You can use `is_class()` to check class by String.
Not sure where to put the method though, perfectly it should be one for all 3 classes.

Reply:
I've simplified the code somewhat for each class. I'm not sure where a generic method could be put so it can be reused across all 3 classes though.